### PR TITLE
C99 use snprintf

### PR DESCRIPTION
**Pull request by richsalz**
## Checklist
Thank you for your contribution. Please answer the following:

- [x] I acknowledge that I am authorized to submit this code under
the terms of the [Apache License](https://www.apache.org/licenses/LICENSE-2.0)
- [ ] All new public APIs are documented
- [ ] All new public functions have tests
- [ ] All new public functions have fuzzing tests

### CODE REVIEW

File: apps/lib/opt.c [hunk 1]
```diff
@@ -149,7 +149,7 @@ char *opt_appname(const char *argv0)
     size_t len = strlen(prog);
 
     if (argv0 != NULL)
-        BIO_snprintf(prog + len, sizeof(prog) - len - 1, " %s", argv0);
+        snprintf(prog + len, sizeof(prog) - len - 1, " %s", argv0);
```
Comment (botovq) [inline apps/lib/opt.c:152]: There's a few of these. Looks like poorly reviewed conversion from what was previously strncpy.

**Comment by richsalz** [comment]
I looked for `snprintf.*”%s”` and replaced them with strclpy.

**Review by botovq — COMMENTED** [review]
This is a terrifying pull request. There's simply too much wrong here, both in the existing code and the code after the change. Clearly much of this code has gone through too many careless approvals and quite possibly has seen no competent review ever.

I believe the only reasonable way to approach this is to split this into file-by-file basis so we can do this right with careful review. Either way this is going to be tedious, but at least doing this in tiny steps has a chance of not introducing too many bugs.

File: crypto/cmp/cmp_client.c [hunk 2]
```diff
@@ -315,23 +315,23 @@ static int poll_for_response(OSSL_CMP_CTX *ctx, int sleep, int rid,
             if (check_after < 0 || (uint64_t)check_after
                 > (sleep ? ULONG_MAX / 1000 : INT_MAX)) {
                 ERR_raise(ERR_LIB_CMP, CMP_R_CHECKAFTER_OUT_OF_RANGE);
-                if (BIO_snprintf(str, OSSL_CMP_PKISI_BUFLEN, "value = %jd",
-                                 check_after) >= 0)
+                if (snprintf(str, OSSL_CMP_PKISI_BUFLEN, "value = %jd",
+			     check_after) >= 0)
```
Comment (botovq) [inline crypto/cmp/cmp_client.c:319]: It's awesome that the -Wformat breakage on macos catches wrong format strings now. This needs an (intmax_t) cast or fiddling with the ugly PRId64 macro.

**Comment by richsalz** [comment]
Yeah, this was a case of tugging on a thread and all of a sudden everything falls apart.

I think the best trade-off is just turn the OpenSSL functions into wrappers around the C99 standard functions.  Closing this.

**Pull request closed without merging**